bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ConnectionController.groovy

```diff
@@ -0,0 +1,47 @@
+/**
+* BigBlueButton open source conferencing system - http://www.bigbluebutton.org/
+*
+* Copyright (c) 2019 BigBlueButton Inc. and by respective authors (see below).
+*
+* This program is free software; you can redistribute it and/or modify it under the
+* terms of the GNU Lesser General Public License as published by the Free Software
+* Foundation; either version 3.0 of the License, or (at your option) any later
+* version.
+*
+* BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+* WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+* PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License along
+* with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+*
+*/
+package org.bigbluebutton.web.controllers
+
+import org.bigbluebutton.api.MeetingService
+import org.bigbluebutton.api.domain.UserSession
+import org.bigbluebutton.api.util.ParamsUtil
+
+class ConnectionController {
+  MeetingService meetingService
+
+  def checkAuthorization = {
+    try {
+      def uri = request.getHeader("x-original-uri")
+      def sessionToken = ParamsUtil.getSessionToken(uri)
```

Comment:
Can you not just get sessionToken directly from `params` like the other controllers? https://github.com/bigbluebutton/bigbluebutton/blob/460c81686819a01e35d35e5eb1594ea87fc08c1c/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy#L1416

Reply:
I tried but did not succeeded when it comes in a wss connection, the SFU case.

Reply:
Strange, but if you tried it then I guess something's different.

Reply:
Well, I'm no nginx expert... Maybe it is possible but I missed on something else.

Reply:
I think this should work for now. We can change this later if we figure out a better way.